cellengine/resources/gate.py

```diff
@@ -303,45 +464,229 @@ def create(
+            y_channel (str): The name of the y channel to which the gate applies.
+            name (str): The name of the gate
+            x (float): The x coordinate of the center point (after the
+                channel's scale has been applied).
+            y (float): The y coordinate (after the channel's scale has been applied).
+            labels (list): Positions of the quadrant labels. A list of four length-2
+                vectors in the order: UR, UL, LL, LR. These are set automatically to
+                the plot corners.
+            skewable (bool): Whether the quadrant gate is skewable.
+            angles (list): List of the four angles of the quadrant demarcations
+            gid (str): Group ID of the gate, used for tailoring. If this is not
+                specified, then a new Group ID will be created. To create a
+                tailored gate, the gid of the global tailored gate must be specified.
+            gids (list): Group IDs of each sector, assigned to ``model.gids``.
+            locked (bool): Prevents modification of the gate via the web interface.
+            tailored_per_file (bool): Whether or not this gate is tailored per FCS file.
+            fcs_file_id (str): ID of FCS file, if tailored per file. Use ``None`` for
+                the global gate in a tailored gate group. If specified, do not
+                specify ``fcs_file``.
+            fcs_file (str): Name of FCS file, if tailored per file. An attempt
+                will be made to find the file by name. If zero or more than one
+                file exists with the name, an error will be thrown. Looking up
+                files by name is slower than using the ID, as this requires
+                additional requests to the server. If specified, do not specify
+                ``fcs_file_id``.
 
+        Returns:
+            A QuadrantGate object.
+
+        Example:
+            ```python
+            cellengine.Gate.create_quadrant_gate(experimentId, x_channel="FSC-A",
+                y_channel="FSC-W", name="my gate", x=160000, y=200000)
+            experiment.create_quadrant_gate(x_channel="FSC-A",
+                y_channel="FSC-W", name="my gate", x=160000, y=200000)
+            ```
+        """
+
+        fcs_file_id = parse_fcs_file_args(
+            experiment_id,
+            tailored_per_file,
+            fcs_file_id,
+            fcs_file,
+        )
+
+        # set labels based on axis scale
+        r = ce.APIClient().get_scaleset(experiment_id, as_dict=True)
+        scale_min = min(x["scale"]["minimum"] for x in r["scales"])
+        scale_max = max(x["scale"]["minimum"] for x in r["scales"])
+
+        if labels == []:
+            labels = [
+                [scale_max, scale_max],  # upper right
+                [scale_min, scale_max],  # upper left
+                [scale_min, scale_min],  # lower left
+                [scale_max, scale_min],  # lower right
```

Comment:
These will be centered at the corners of the plots and cut off. Instead they need to be nudged inward in both directions. Unfortunately that nudge depends on the gate label font size and the plot size. Below is what we use in CellEngine for the 291-pixel plot that we use in gating:

```
{x: xmax - 32, y: ymax + 16, region: QuadrantSector.UR},
{x: xmin + 40, y: ymax + 16, region: QuadrantSector.UL},
{x: xmin + 40, y: ymin - 15, region: QuadrantSector.LL},
{x: xmax - 32, y: ymin - 15, region: QuadrantSector.LR}
```

Those are pixels, so you need to subtract the scale min from the max, map that range to [0, 290], nudge, and then go back to the scale range.

Reply:
Let me know if this solution is what you're looking for; it provides labels at the same locations as CE

Reply:
Haven't tested it, but code looks correct. 👍

Reply:
I've tested it on CE, but it's probably worth a second look. Thanks!